tests/test_data.py

```diff
@@ -1323,6 +1323,52 @@ def test_inversions(self, gslice):
             assert np.all(np.array(res) == vdat[tuple(sl)])
             assert res.shape == vdat[tuple(sl)].shape
 
+    @pytest.mark.parallel(mode=4)
+    def test_setitem_shorthands(self):
+        # Test setitem with various slicing shorthands
+        nx = 8
+        ny = 8
+        nz = 8
+        shape1 = (nx, ny)
+        shape2 = (nx, ny, nz)
+        grid1 = Grid(shape=shape1, dtype=np.float32)
+        x, y = grid1.dimensions
+        glb_pos_map = grid1.distributor.glb_pos_map
+        grid2 = Grid(shape=shape2, dtype=np.float32)
+        f1 = Function(name='f1', grid=grid1)
+        f2 = Function(name='f2', grid=grid1)
+        g = Function(name='g', grid=grid2)
+
+        dat1 = np.arange(64, dtype=np.int32).reshape(grid1.shape)
+
+        f1.data[:] = dat1
+        f2.data[:] = f1.data[:]
+        assert np.all(f2.data[:] == f1.data[:])
+
+        g.data[0, :, :] = dat1
+        f1.data[:] = g.data[0, ::-1, ::-1]
+        result = np.array(f1.data[:])
+        if LEFT in glb_pos_map[x] and LEFT in glb_pos_map[y]:
```

Comment:
These lines can be done parametrically as well, but its pretty clear this way, not sure if it is gonna be better parametrically